### PR TITLE
New dragging

### PR DESCRIPTION
**Pull request by vxsx**
## Changes
- removed droppable, no need for weird interconnections now, updated jquery ui used.
- new look for things
- fixed bug with isAllowed not returning correct state (though may be not all cases)
- changed clone helper to be custom since there's a huge performance loss when cloning deeply nested list
- changed `cms-draggables` lists manipulations from `show/hide` to classes because `show/hide` is very slow on big sets
- success animation is now longer
- when dragging plugins in and out their collapsible state now updates properly
- fixed cancelling by pressing esc to not cancel previous interaction and to only work while dragging

## Not solved yet
- there is some juggling of the line placeholder on deeply nested plugins while draggig
- performance is a bit sad on hugely nested (30-40) lists, esp on mobile

## screenshots for new shiny things

valid position for the drop (dragging simple single plugin)
screen shot 2015-09-10 at 17 21 33 3

dragging bunch of plugins (has children basically)
screen shot 2015-09-10 at 17 21 25 3

successful move fadeout
screen shot 2015-09-10 at 17 21 29 3

invalid position for the drop
screen shot 2015-09-10 at 17 21 46 3

drop onto an empty placeholder (possible)
screen shot 2015-09-10 at 17 21 56 3

drop onto an empty placeholder (not possible)
screen shot 2015-09-10 at 17 31 28

### CODE REVIEW

File: cms/static/cms/js/modules/cms.plugins.js [hunk 1]
```diff
@@ -575,11 +575,12 @@
                 });
 
                 // prevent propagnation
-                nav.on(this.click + ' dblclick', function (e) {
+                nav.on(this.click + ' click dblclick', function (e) {
```
Comment (FinalAngel) [inline cms/static/cms/js/modules/cms.plugins.js]: what about cms event namespacing here?

File: cms/static/cms/js/modules/cms.plugins.js [hunk 2]
```diff
@@ -781,8 +782,11 @@
                 }
 
                 // attach events to draggable
-                this.ui.draggable.find('> .cms-dragitem-collapsable').on(this.click, function () {
+                this.ui.draggable.find('> .cms-dragitem').on('click.cms', function () {
```
Comment (FinalAngel) [inline cms/static/cms/js/modules/cms.plugins.js]: better click.cms.plugin

File: cms/static/cms/js/modules/cms.structureboard.js [hunk 3]
```diff
@@ -11,7 +11,8 @@
         function actualizeEmptyPlaceholders() {
             placeholders.each(function () {
                 var placeholder = $(this);
-                if (placeholder.find('> .cms-draggables').children().not('.cms-is-dragging').length) {
+                if (placeholder
+                        .find('> .cms-draggables').children('.cms-draggable:not(.cms-draggable-is-dragging)').length) {
```
Comment (FinalAngel) [inline cms/static/cms/js/modules/cms.structureboard.js]: also break at .children, makes it more readable

File: cms/static/cms/js/modules/cms.structureboard.js [hunk 4]
```diff
@@ -19,6 +20,19 @@
             });
         }
 
+        function actualizePluginsCollapsibleStatus(els) {
+            els.each(function () {
+                var childList = $(this);
```
Comment (FinalAngel) [inline cms/static/cms/js/modules/cms.structureboard.js]: variables on top of els.each

File: cms/static/cms/sass/components/_structureboard.scss [hunk 5]
```diff
@@ -215,19 +219,115 @@
...
+        display: none;
+    }
+    .cms-dragarea-empty {
+        .cms-droppable {
+            transform: translateY(0) !important;
+            color: darken($gray-lighter, 15%);
+            height: auto;
+            box-shadow: none;
+            margin: 0;
+            background-color: mix(white, $color-primary, 90%);
```
Comment (FinalAngel) [inline cms/static/cms/sass/components/_structureboard.scss]: ordering

File: cms/static/cms/sass/components/_structureboard.scss [hunk 6]
```diff
@@ -215,19 +219,115 @@
...
         transform: translateZ(0);
+        z-index: z(default);
+        width: $structure-draggable-width-while-dragging !important;
+        border-radius: 0;
+        height: $structure-dragitem-fullheight !important;
+        .cms-dragitem {
+            border-radius: 0;
+            border: none;
+            padding-left: $structure-dragarea-padding-horizontal;
+            color: $white !important;
```
Comment (FinalAngel) [inline cms/static/cms/sass/components/_structureboard.scss]: also ordering ;)

File: cms/static/cms/sass/components/_structureboard.scss (hunk 5 above)
Comment (vxsx) [inline cms/static/cms/sass/components/_structureboard.scss]: actually yeah, i'm gonna csscomb

File: cms/static/cms/js/modules/cms.plugins.js (hunk 1 above)
Comment (vxsx) [inline cms/static/cms/js/modules/cms.plugins.js]: will add. although i think "cleanup plugins" is somewhere in the pipeline

**Comment by FinalAngel** [comment]
LGTM

**Comment by FinalAngel** [comment]
LGTM :+1: :+1:

**Pull request merged**